server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/toolShared.ts

```diff
@@ -51,21 +52,73 @@ export enum OutputKind {
     Json = 'json',
 }
 
+/**
+ * Checks if a path has already been approved
+ * @param path The path to check
+ * @param approvedPaths Set of approved paths
+ * @returns True if the path or any parent directory has been approved
+ */
+export function isPathApproved(filePath: string, approvedPaths?: Set<string>): boolean {
+    if (!approvedPaths || approvedPaths.size === 0) {
+        return false
+    }
+
+    // Normalize path separators for consistent comparison
+    const normalizedPath = filePath.replace(/\\/g, '/')
+
+    // Check if the exact path is approved (try both original and normalized)
+    if (approvedPaths.has(filePath) || approvedPaths.has(normalizedPath)) {
+        return true
+    }
+
+    // Check if any parent directory is approved
+    let currentPath = normalizedPath
+    const rootDir = path.parse(filePath).root.replace(/\\/g, '/')
+
+    while (currentPath !== rootDir) {
```

Comment:
could you use something like [this](https://github.com/aws/language-servers/blob/a7bfc1a0da42392a26fe07ba5918c1d7d761de86/core/aws-lsp-core/src/util/workspaceUtils.ts#L100) here?